### ISSUE TITLE
Inconsistency outputs of Separable and Depthwise Convolution layer between TensorFlow and CNTK/Theano

### ISSUE THREAD

**Issue by hvpham** [keras-team/keras#10672]
I am using the NASNetLarge/NASNetMobile and the MobileNetV2 on the lastest Keras version (github build). I am testing with all three backends and I see the differences between TensorFlow and Theano/CNTK.

By plotting the normalized mean squared error it seem to started from layer separable_conv_1_reduction_left1_reduce_6 (NASNetLarge), separable_conv_1_reduction_left1_stem_2 (NASNetMobile), and the expanded_conv_depthwise (MobileNetV2).

I looked at the source code and saw that both Theano and CNTK does not have native support for Separable Convolution layer or DepthWise Convolution layer, only TensorFlow has these functionalities. In recent versions (2.2.0 for Separable and 2.1.5 for Depthwise), these were added to Theano and CNTK  by using normal conv2D.

I am just not sure if this means that the implementation in cnkt_backend.py and theano_backend.py are correct (they are consistent with each other) or the implementation in TensorFlow native code is correct.

Attached are the plots that I was talking about (for NASNetLarge, NASNetMobile, and MobileNetV2 respectively). These are the normalized mean square differences of the outputs at each layer between Tensorflow and CNTK (We have the same graph for Tensorflow vs Theano since Theano agrees with CNTK). As you can see toward the end, the prediction is very different between the two backends (I also included the prediction result for each combination to so the severity of inconsistency)

![nmse_val_00000317_tensorflow_cntk_nasnetlarge](https://user-images.githubusercontent.com/9290537/42706864-3c97b7ee-86a6-11e8-9849-18bbe50e5c6a.png)

![image](https://user-images.githubusercontent.com/9290537/42707436-2f241ace-86a8-11e8-80fa-8aa8c22888a2.png)

![nmse_val_00001913_tensorflow_cntk_nasnetmobile](https://user-images.githubusercontent.com/9290537/42706889-48b6f620-86a6-11e8-9c30-3a245b5622e9.png)

![image](https://user-images.githubusercontent.com/9290537/42707529-7a6c3cd2-86a8-11e8-86fc-fc97266f5d58.png)

![nmse_val_00001996_tensorflow_cntk_mobilenetv2](https://user-images.githubusercontent.com/9290537/42706905-61294046-86a6-11e8-82ac-938ba86b86ac.png)

![image](https://user-images.githubusercontent.com/9290537/42707569-9b59d378-86a8-11e8-9626-a8d429a379be.png)
